packages/cli/generation/ir-generator/src/filtered-ir/IrGraph.ts

```diff
@@ -388,6 +398,24 @@ export class IrGraph {
                 }
             }
 
+            for (const [endpointId, queryParametersNode] of Object.entries(this.queryParameters)) {
+                if (!this.endpointsNeededForAudience.has(endpointId)) {
+                    continue;
+                }
+                const parametersForEndpoint = new Set<string>();
+                for (const audience of this.audiences.audiences) {
+                    const parametersByAudience = queryParametersNode.parametersByAudience[audience];
+                    if (parametersByAudience != null) {
+                        parametersForEndpoint.forEach((parameter) => {
```

Comment:
Should this be the following?
```suggestion
                        parametersByAudience.forEach((parameter) => {
```